lib/run-script.js

```diff
@@ -46,12 +57,18 @@ class RunScript extends BaseCommand {
       this.list(args).then(() => cb()).catch(cb)
   }
 
-  async run (args) {
-    const path = this.npm.localPrefix
-    const event = args.shift()
+  execWorkspaces (args, filters, cb) {
+    if (args.length)
+      this.runWorkspaces(args, filters).then(() => cb()).catch(cb)
+    else
+      this.listWorkspaces(args, filters).then(() => cb()).catch(cb)
+  }
+
+  async run (_args, { path = this.npm.localPrefix, pkg } = {}) {
+    const [event, ...args] = _args
```

Comment:
Why was this changed from the unshift approach?  It is way more clear there what's going on and doesn't require a variable that's slightly differently named just to juggle that around

Reply:
> It is way more clear there what's going

ah, that's an interesting comment since from my pov the shift approach was way **less** clear, I guess it might be more of a background perspective coming from the front-end/react world deconstruction has been the norm for waaay to long now 😅 I guess my point is when you're in the default state of mind where favoring immutable APIs over mutable ones is the norm, mutable APIs are **very** confusing

but to answer to your question:

> Why was this changed from the unshift approach?

that's actually necessary since the function is called multiple times now and `shift()` is going to mutate the original array

Reply:
Can we make this `async run ([event, ...args], ...)` then?